packages/hs-react-ui/src/components/Table/Table.stories.tsx

```diff
@@ -182,3 +223,136 @@ storiesOf('Table', module).add(
+
+storiesOf('Table', module).add(
+  'Groups',
+  () => {
+    const [rows, setRows] = useState(sampleGroupData);
+
+    const onSelect = (index: number, groupIndex = 0, selected: boolean) => {
+      const newRows: Array<SampleSelectionCellType[]> = [];
+      rows.forEach((grp: SampleSelectionCellType[]) => {
+        newRows.push([...grp]);
+      });
+      // const newRows = [...rows];
+      newRows[groupIndex][index].selected = !selected;
+      setRows(newRows);
+    };
+
+    const selectAll = (evt: SyntheticEvent) => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const currentlyChecked = evt.target.checked;
+      const newRows: Array<Array<RowProps>> = [];
+      rows.forEach((group: Array<RowProps>) => {
+        newRows.push(
+          group.map((row: RowProps) => {
+            return { ...row, selected: currentlyChecked };
+          }),
+        );
+      });
+      setRows(newRows);
+    };
+
+    const SelectAllCell = () => {
+      let totalSelected = 0;
+      let totalCheckboxesAccumulator = 0;
+      rows.forEach(groupRows => {
+        groupRows.forEach((row: columnTypes) => {
+          if (groupRows.isGroupLabel) return;
+          if (row.selected) totalSelected++;
+          totalCheckboxesAccumulator++;
+        });
+      });
+      const allChecked = totalSelected === totalCheckboxesAccumulator;
+
+      // TODO: don't use pointer-events to control if a column is sortable - it should be checked
+      // within the sorting listeners so that `sortable` doesn't need to be passed here just to give
+      // it pointer-events
+      return (
+        <Table.HeaderCell sortable>
+          <Checkbox
+            checkboxType={allChecked ? CheckboxTypes.check : CheckboxTypes.neutral}
+            checked={Boolean(totalSelected)}
+            onClick={selectAll}
+          />
+        </Table.HeaderCell>
+      );
+    };
+
+    const SelectionCell = ({
+      index,
+      selected,
+      reachedMinWidth,
+      groupIndex,
+    }: {
+      index: number;
+      selected: boolean;
+      reachedMinWidth?: boolean;
+      groupIndex?: number;
+    }) => (
+      <Table.Cell>
+        <Checkbox
+          onClick={() => onSelect(index, groupIndex, selected)}
+          checkboxType={CheckboxTypes.check}
+          checked={selected}
+        >
+          {reachedMinWidth ? 'Select for download' : ''}
+        </Checkbox>
+      </Table.Cell>
+    );
+
+    const EmptyCell = () => <Table.Cell />;
+
+    const NotesCell = ({ notes }: { notes: string }) => (
+      <Table.Cell>
+        <NoteField onChange={() => console.log()} rows={3} value={notes} />
+      </Table.Cell>
+    );
+
+    const sampleColumns: { [index: string]: any } = {
+      selection: {
+        name: '',
+        headerCellComponent: SelectAllCell,
+        width: text('Selection width', '2rem'),
+        cellComponent: SelectionCell,
+        sortable: false,
+        groupCellComponent: EmptyCell,
```

Comment:
Is an empty cell required for the groupCellComponent if a cellComponent is defined?

Reply:
The intention is if `cellComponent` is defined and `groupCellComponent` is not, then `cellComponent` is used. `GroupCellComponent` should be used to override the `CellComponent` for the group 'header.' I meant for it to work like this, but just checked the implementation and that needs corrected.

Reply:
If EmptyCell is just Table.Cell, couldn't you just pass that through?

Reply:
Oh nevermind - EmptyCell is a Table.Cell with nothing in it. cool.